HIP/hip-646.md

```diff
@@ -0,0 +1,90 @@
+---
+hip: 646
+title: Fungible Token Metadata Field
+author: May Chan (@rocketmay)
+working-group: Ashe Oro (@Ashe-Oro)
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Review
+created: 2023-01-04
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/627
+updated: 2023-01-04
+---
+
+## Abstract
+
+This HIP proposes the addition of the metadata field to Fungible Tokens (FT), taking after the Non-Fungible Token (NFT) metadata field which was added in HIP-17.
+
+## Motivation
+
+Similar to NFT's on the network, the fungible token implementation is largely functional. There is a need for a token creator to specify additional "metadata" about the token. The data and formatting is explored in HIP-405 but there is no field suitable for storing the link to this metadata. 
+
+The most 'logical' field proposed to store this data is the existing 'memo' field, however a dedicated metadata field is more appropriate.
+
+## Rationale
+
+The success of HIP-412 in standardizing the NFT space on Hedera demonstrates the potential of standardized metadata functionality for FT's. Wallets, explorers and other applications can display relevant supplementary information about tokens to users in a decentralized fashion.
+
+## User stories
+
+As a token creator, I want a platform-standardized method to host information about my token in a decentralized fashion, allowing applications to query data such as token icon and website.
+
+As a wallet or explorer, I want to query a mirror node to obtain the metadata for any fungible token, similar to current non-fungible token calls, to simplify calls and implementation.
+
+As a developer, I want to store the metadata uri in a dedicated field, to avoid ambiguity of using an overloaded field such as 'memo'.
+  
+## Specification
+
+## HAPI Changes
+
+```
+   // Retrieves the metadata of a token
+   rpc getTokenInfo (Query) returns (Response);
+```
+
+### TokenMintTransactionBody
+Property `metadata` is updated to include all tokens.
+```
+message TokenMintTransactionBody {
+    TokenID token = 1; // The token for which to mint tokens. If token does not exist, transaction results in INVALID_TOKEN_ID
+!   uint64 amount = 2; // Applicable to tokens of type FUNGIBLE_COMMON. The amount to mint to the Treasury Account. Amount must be a positive non-zero number represented in the lowest denomination of the token. The new supply must be lower than 2^63.
++   repeated bytes metadata = 3; // Applicable to **all tokens**, previously was only NON_FUNGIBLE_UNIQUE. A list of metadata that are being created. Maximum allowed size of each metadata is 100 bytes
```

Comment:
What's the motivation for a list of metadata?

Reply:
Just a copy-paste error, good catch thank you